### PR TITLE
♻️ refactor eventListeners inside useEffect for stable pointers

### PR DESCRIPTION
**Pull request by mayteio**
Addresses #20.

### CODE REVIEW

File: src/use-localstorage.ts [hunk 1]
```diff
@@ -48,19 +58,16 @@ export function useLocalStorage<TValue = string>(key: string, initialValue?: TVa
...
 
   useEffect(() => {
-    // The custom storage event allows us to update our component 
+    // The custom storage event allows us to update our component
     // when a change occurs in localStorage outside of our component
-    window.addEventListener(
-      LocalStorageChanged.eventName,
-      (e: any) => onLocalStorageChange(e as LocalStorageChanged<TValue>)
-    );
+    const listener = (e: Event) => onLocalStorageChange(e as LocalStorageChanged<TValue>);
```
Comment (mayteio) [inline src/use-localstorage.ts]: A stable pointer is set inside the `useEffect`. This renders the `useCallback` wrapping `onLocalStorageChange` function redundant.

File: src/use-localstorage.ts [hunk 2]
```diff
@@ -72,20 +79,13 @@ export function useLocalStorage<TValue = string>(key: string, initialValue?: TVa
     }
 
     return () => {
-      window.removeEventListener(
-        LocalStorageChanged.eventName,
-        (e: any) => onLocalStorageChange(e as LocalStorageChanged<TValue>)
-      );
-      window.removeEventListener('storage', e => onLocalStorageChange(e));
+      window.removeEventListener(LocalStorageChanged.eventName, listener);
```
Comment (mayteio) [inline src/use-localstorage.ts]: Now, when the listener is passed to `removeEventListener` it is referentially equal and actually removes it.

**Comment by iamsolankiamit** [comment]
@mayteio can you resolve the conflicts, other than that all looks good to me. LGTM

**Comment by mayteio** [comment]
All done @iamsolankiamit. Cheers!

**Comment by mayteio** [comment]
Thanks @iamsolankiamit, when do you think you might cut a new release? Cheers

**Comment by iamsolankiamit** [comment]
@mayteio released under https://github.com/rehooks/local-storage/releases/tag/v2.2.2

**Pull request merged**